src/redux/reducers/compare.ts

```diff
@@ -0,0 +1,412 @@
+import {PayloadAction, TaskAbortError, createSelector, createSlice, isAnyOf} from '@reduxjs/toolkit';
```

Comment:
@devcatalin I went for a feature slice but want to check whether you dislike it and how we can structure this big file into smaller files.